### ISSUE TITLE
Implementation Components

### ISSUE THREAD

**Issue by AlexLacson** [LindsayXX/NEM#6]
### **General**

- [x] Data reading/processing (PyTorch has package to download some standard data) 

- [ ]  Argument parser

- [x] KL divergence (slightly different between AGE and IntroVAE/VAE)

- [x] Z-sampling

- [x] Training batch loader

- [x] Image generator

- [ ] Quantitative metrics

- [x] L2 norm

### **AGE**

- [x] Training procedure 

- [x] 32x32 Encoder 

- [x] 32x32 Generator

- [x] 128x128 Encoder

- [x] 128x128 Generator 

- [x] Loss function (Based on KL divergence)

### **IntroVAE**

- [x] Training procedure 

- [x] 128x128 Encoder

- [x] 128x128 Generator 

- [x] Loss function (Based on KL divergence)

### **Vanilla VAE**

- [x] Training procedure 

- [x] Loss function (Based on KL divergence)
